homeassistant/components/kaleidescape/sensor.py

```diff
@@ -0,0 +1,167 @@
+"""Sensor platform for Kaleidescape integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+
+from .const import DOMAIN as KALEIDESCAPE_DOMAIN
+from .entity import KaleidescapeEntity
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from kaleidescape import Device as KaleidescapeDevice
+
+    from homeassistant.config_entries import ConfigEntry
+    from homeassistant.core import HomeAssistant
+    from homeassistant.helpers.entity_platform import AddEntitiesCallback
+    from homeassistant.helpers.typing import StateType
+
+
+@dataclass
+class BaseEntityDescriptionMixin:
+    """Mixin for required descriptor keys."""
+
+    value_fn: Callable[[KaleidescapeDevice], StateType]
+
+
+@dataclass
+class KaleidescapeSensorEntityDescription(
+    SensorEntityDescription, BaseEntityDescriptionMixin
+):
+    """Describes Kaleidescape sensor entity."""
+
+
+SENSOR_TYPES: tuple[KaleidescapeSensorEntityDescription, ...] = (
+    KaleidescapeSensorEntityDescription(
```

Comment:
Should any of these have a state class?
https://developers.home-assistant.io/blog/2021/08/16/state_class_total?_highlight=state&_highlight=class#state-classes
https://developers.home-assistant.io/docs/core/entity/sensor#properties

It looks like some of them should have an entity category
https://developers.home-assistant.io/blog/2021/10/26/config-entity?_highlight=diagnostic#entity-categories

Should any of these have a device class?
https://developers.home-assistant.io/docs/core/entity/sensor?_highlight=device&_highlight=class#available-device-classes

It looks like some of these need a `native_unit_of_measurement`
https://developers.home-assistant.io/docs/core/entity/sensor#properties

Reply:
Looks like state and device classes don't apply to any of them. I did the add the diagnostic category to most of them, and added the percentage native_unit_of_measurement to those that applied.

Reply:
Ok fixed in the library with this commit https://github.com/SteveEasley/pykaleidescape/commit/753a49c72fe8b632e8dbb861ecef7e198f749e13